app/js/directives.js

```diff
@@ -432,28 +432,38 @@ angular.module('myApp.directives', ['myApp.filters'])
       templateUrl: templateUrl('message_attach_contact')
     }
   })
-  .directive('myMessageWebpage', function (AppWebPagesManager, AppPhotosManager) {
+  .directive('myMessageWebpage', function (AppWebPagesManager, AppPhotosManager, Storage) {
     return {
       scope: {
         'media': '=myMessageWebpage',
         'messageId': '=messageId'
       },
       templateUrl: templateUrl('message_attach_webpage'),
       link: function ($scope) {
-        $scope.openPhoto = AppPhotosManager.openPhoto
-        $scope.openEmbed = function ($event) {
-          if ($scope.media.webpage &&
-            $scope.media.webpage.embed_url) {
-            AppWebPagesManager.openEmbed($scope.media.webpage.id, $scope.messageId)
-            return cancelEvent($event)
-          }
-        }
+        Storage.get('disable_webpage_preview').then(function (previewDisabled) {
+          $scope.previewDisabled = !!previewDisabled;
```

Comment:
@auchri Not quite. `!!previewDisabled` casts `previewDisabled` to a Boolean by first negating it (thereby casting it to a Boolean) and then invert the value (that is, falsy values become `false` and truthy `true`).